recipes/sdl_image/all/conanfile.py

```diff
@@ -73,68 +66,104 @@ def config_options(self):
-    def _configure_cmake(self):
-        cmake = CMake(self)
-        cmake.definitions["BMP"] = self.options.bmp
-        cmake.definitions["GIF"] = self.options.gif
-        cmake.definitions["IMAGEIO"] = self.options.get_safe("imageio")
-        cmake.definitions["JPG"] = self.options.with_libjpeg
-        cmake.definitions["LBM"] = self.options.lbm
-        cmake.definitions["PCX"] = self.options.pcx
-        cmake.definitions["PNG"] = self.options.with_libpng
-        cmake.definitions["PNM"] = self.options.pnm
-        cmake.definitions["SVG"] = self.options.svg
-        cmake.definitions["TGA"] = self.options.tga
-        cmake.definitions["TIF"] = self.options.with_libtiff
-        cmake.definitions["WEBP"] = self.options.with_libwebp
-        cmake.definitions["XCF"] = self.options.xcf
-        cmake.definitions["XPM"] = self.options.xpm
-        cmake.definitions["XV"] = self.options.xv
-        # TODO: https://github.com/bincrafters/community/pull/1317#pullrequestreview-584847138
-        cmake.definitions["TIF_DYNAMIC"] = self.options["libtiff"].shared if self.options.with_libtiff else False
-        cmake.definitions["JPG_DYNAMIC"] = self.options["libjpeg"].shared if self.options.with_libjpeg else False
-        cmake.definitions["PNG_DYNAMIC"] = self.options["libpng"].shared if self.options.with_libpng else False
-        cmake.definitions["WEBP_DYNAMIC"] = self.options["libwebp"].shared if self.options.with_libwebp else False
-        cmake.definitions["SDL_IS_SHARED"] = self.options["sdl"].shared
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
-        cmake.configure(build_folder=self._build_subfolder)
-        return cmake
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version],
+            destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["SDL_IMAGE_SRC_DIR"] = self.source_folder.replace("\\", "/")
+        tc.variables["BMP"] = self.options.bmp
+        tc.variables["GIF"] = self.options.gif
+        tc.variables["IMAGEIO"] = self.options.get_safe("imageio")
+        tc.variables["JPG"] = self.options.with_libjpeg
+        tc.variables["LBM"] = self.options.lbm
+        tc.variables["PCX"] = self.options.pcx
+        tc.variables["PNG"] = self.options.with_libpng
+        tc.variables["PNM"] = self.options.pnm
+        tc.variables["SVG"] = self.options.svg
+        tc.variables["TGA"] = self.options.tga
+        tc.variables["TIF"] = self.options.with_libtiff
+        tc.variables["WEBP"] = self.options.with_libwebp
+        tc.variables["XCF"] = self.options.xcf
+        tc.variables["XPM"] = self.options.xpm
+        tc.variables["XV"] = self.options.xv
+        # TODO: https://github.com/bincrafters/community/pull/1317#pullrequestreview-584847138
+        tc.variables["TIF_DYNAMIC"] = self.dependencies["libtiff"].options.shared if self.options.with_libtiff else False
```

Comment:
`*_DYNAMIC` variables ([`*_SHARED` in upstream](https://github.com/libsdl-org/SDL_image/blob/18ca89d0f577476f2340c6ab18a312c314b1e696/CMakeLists.txt#L120) means dynamically loading the library using `dlopen`.

In upstream code, if a 3rd party library is dynamically loaded then the library must be shared.
If the 3rd party library is not dynamically loaded, then the library can be shared or static.

I'm repeating this here because since conan adds these 3rd party libraries to `requires` in `package_info` and those 3rd party libraries are thus added to the link flags, it makes more sense to always disable `*_DYNAMIC` here imho.

But this is out of scope for this pr.

Reply:
Yes I remember to also have raised this fact in the first PR of this recipe. So currently in this recipe, if a direct dependency of sdl_image is shared, it will be dynamically loaded by sdl_image at runtime, but there is not option to link to it instead.